### ISSUE TITLE
DataTable: rows disappear when data changes

### ISSUE THREAD

**Issue by ArtBat123** [primefaces/primevue#4994]
### Describe the bug

When using virtualscroll, rows disappear in the table when we scroll to the middle of the table and reduce the amount of data in it. When using horizontal scrolling, data appears.
Browser- Firefox 106 !!
https://github.com/primefaces/primevue/assets/79512478/2304c0d9-38dc-4119-94d0-4af43f2b9151

### Reproducer

https://codesandbox.io/s/primevue-demo-forked-nml37x?file=/src/App.vue

### PrimeVue version

3.45.0

### Vue version

3.x

### Language

ES6

### Build / Runtime

Vite

### Browser(s)

Firefox 106

### Steps to reproduce the behavior

1.Scroll to the middle
2. Click on the "1 row" button to update the data

### Expected behavior

The line must be visible

**Comment by mertsincan** [comment]
Hi,

Thanks a lot for your report! I set a milestone for it. We'll check before the milestone is released. Also, your sample is not working. Could you please update it? Thank you!

**Comment by ArtBat123** [comment]
Thanks for the answer. Updated link to example. https://codesandbox.io/s/primevue-demo-forked-z7cnvy?file=/src/App.vue

**Comment by ArtBat123** [comment]
The transform: translate3d() property does not update its value. In this case, Firefox does not fire the onScroll event. It might be worth updating the transform: translate3d() property when updating data.

**Comment by ArtBat123** [comment]
@tugcekucukoglu Yes, sure. Try this link https://stackblitz.com/edit/saortp?file=src%2FApp.vue
